### ISSUE TITLE
Problem create VPN with TAG subscribingVpc = Yes

### ISSUE THREAD

**Issue by SergiMajo** [PaloAltoNetworks/aws-transit-vpc#9]
Hi team, 

After setting tag to Yes, the VPNs do not come up (we have waited about 1 hour).
Any clues?

Thanks

**Comment by SergiMajo** [comment]
Hi, 

I have full access to everything in the permissions, but could it still be a permissions problem? doesn't seem to write in dynamodb  for create vpn.

regards,

**Comment by narayan-iyengar** [comment]
Did you run the InitializesubscribeAccount template?

Thanks,
--
/narayan

**Comment by SergiMajo** [comment]
Yeah. the script creates without problems a second machine but never creates a vpn between vpc's. 
regards

**Comment by Vinch157** [comment]
Any news about this? I'm experiencing the same problem. The InitializesubscribeAccount template has been run but when changing the tag, no VPN is created.

**Comment by rr128** [comment]
We are having the same problem.  Our existing VPC does not have an existing VGW or IGW.

**Comment by narayan-iyengar** [comment]
Have  you run the InitializeTransitaAccount template in the hub?

If you have and still tunnels don't come up, try to log into one of the firewalls using the user/pass described in the doc. 
If that fails, then chances are bootstrapping has failed.

Make sure your bootstrap bucket is properly organized and that the bucket is in the same region as the hub transit (where you deployed the initialize template)

**Comment by Vinch157** [comment]
Hello,

Yes, the InitializeTransitAccount has been launched successfully.
In fact, I can "attach" a subscribing VPC by the Option 1 described in the document.
However I have another VPC which already exists and I want to attach it via Option 3. So, I just added a tag (key: subscribingVpc and Value: Yes). It seems the "CloudTrailLambda" function is triggered (that's what I see in CloudWatch), but in the end, the VGW is not attached to the VPC. The VPC does not have any IGW nor VGW previously configured. It just has one subnet, and there is no instance in it. It is basically a brand new VPC with only one subnet in it. Are there specific conditions to be matched by the VPC that has to be attached with Option 3?

Thank you in advance.

**Comment by rr128** [comment]
Same problem here.

**Comment by Vinch157** [comment]
Hello, I just tried again and this time, it has worked!! I think I wrote the key "SubscribingVpc" instead of "subscribingVpc". Maybe the case is important for the tag?

**Comment by rr128** [comment]
I always had subscribingVpc and it never worked.  Did you put in "yes" or "YES" as the value?

**Comment by narayan-iyengar** [comment]
I believe it is all caps…

--
/narayan

**Comment by rr128** [comment]
Doesn't work for me when i put an existing VPC without an IGW and VGW and tag it with subscribingVPC = YES

**Comment by narayan-iyengar** [comment]
Ok. Will have to check the code...Will update.


--
Thanks,
/narayan
________________________________

**Comment by narayan-iyengar** [comment]
The tag needs to be susbcribingVpc=YES/Yes/yes

It is listed in the documentation:

documentation/solution_overview.md:14

--
/narayan

**Comment by Vinch157** [comment]
Hello,

Just a note from my experience: you need to wait quite a long time for the PAGroup58 to be fired up. At least 10 minutes + time to initialize the FWs + time to bring up the VPNs. So, in total 15 - 20 minutes. A small hint: check CloudWatch if you see the Lambda function be triggered.

I tried to tag another VPC as subscribing VPC which has already an IGW attached to it (no VGW but well one IGW) and it works! The VPN are coming up. Is this supported? I read in the document that it was normally not supported. 

I also tried something else: I want that a specific route is injected via the VGW, not the default route as mentionned in the documentation. If you change the redistribution profile accordingly in the FW it is working. So finally, I can have a Subscribing Vpc having a default route pointing to an IGW (default route to internet) and a specific route (to another Subscribing VPC for example) pointing to the VGW going to the firewall. Is this also something that is normally supported? (no negative side effect?)

**Comment by rr128** [comment]
When you tagged an existing VPC, what is from another account?

**Comment by narayan-iyengar** [comment]
You can deploy the hub and spoke in the same aws account or have transit in one and the spoke in the other.


--
Thanks,
/narayan
________________________________

**Comment by rr128** [comment]
When I try to tag an existing VPC from another account, it does not associate with the transit VPC.  The existing vpc does not have a VGW or IGW.  I am using the initializeSubscriberAccount.json to connect to spoke VPC to the transit VPC.

**Comment by narayan-iyengar** [comment]
And you have specified the account numbers when deploying the respective templates?


--
Thanks,
/narayan
________________________________

**Comment by rr128** [comment]
Yes, I’m putting the transit account number within the subscriber template.  Also, I have the S3 bucket within the subscriber account.  Is that correct?

**Comment by narayan-iyengar** [comment]
For lambda functions. Yes.


--
Thanks,
/narayan
________________________________

**Comment by originalwarby** [comment]
Hi Vinch157.  Using both an IGW and a VGW in a subscribing (spoke) VPC will work from a routing perspective.  And this is published as community supported so your changes won't change the support.

But you are allowing traffic out to the internet without any inspection.  This means the firewalls can't look for things like data exfiltration or connections to known malicious URLs etc.  So technically this should work but we wouldn't consider this as secure.

Best practices would be to default route to the VGW and then route through the transit VM-Series firewalls to the IGW in the transit VPC.  This gets you secured Internet access without hair-pinning back to an on-prem firewall.

**Comment by rr128** [comment]
i see the following error within the steps function when i try to tag a vpc

"error": "States.Runtime",
  "cause": "An error occurred while executing the state 'ChoiceState' (entered at the event id #19). Unable to apply Path transformation to null input."

**Comment by rr128** [comment]
Actually, this seems to be the error that is causing the problem:

Error in publishToSns(), Error: An error occurred (AccessDenied) when calling the AssumeRole operation: User: arn:aws:sts::XXXXXXXX:assumed-role/SubscriberLambdaExecutionRole-Test-Spoke-VPC/createVpnConnection-Test-Spoke-VPC is not authorized to perform: sts:AssumeRole on resource: arn:aws:iam::XXXXXXXXX:role/TransitAssumeRole-Transit-VPC

**Comment by narayan-iyengar** [comment]
Good catch.

When launching the transit template make sure the subscriber account number is correct.

When launching the subscriber account template make sure the following are also correct:
[cid:image001.png@01D3E216.C18345C0]

The first two are output of the transit initializer template.

In your subscriber template deployment, you can check the stack deployment parameters and see if there is no typo.



--
/narayan

**Comment by rr128** [comment]
thanks.  i don't see the picture you attach.

**Comment by narayan-iyengar** [comment]
It is in the deployment guide…in the documentation folder in the repo.

Hopefully you have been following that

--
/narayan

**Comment by rr128** [comment]
Got it to work.  Thanks for all the help.

**Comment by narayan-iyengar** [comment]
Awesome. Can you please bupdate the thread with what you did so other folks can benefit as well. Thanks.


--
Thanks,
/narayan
________________________________

**Comment by narayan-iyengar** [comment]
I have noticed sometimes it takes a while.

Are they still not up? On the fw are they showing down?


--
/narayan

**Comment by Vinch157** [comment]
Hello @originalwarby,

Yes, you are completely right. 
But in my setup I'm in fact trying to merge the "aws-transit-vpc" and the "aws-elb-autoscaling". 
In this setup the Subscriber VPC is a spoke that is linked to two hubs: 

- one southbound hub is the transit VPC (to be able to connect to other Subscriber VPCs or to the on-premises server (a VPN connection is then made between the firewalls of the Transit VPC and on premises firewall)).

- one northbound hub which is an autoscaling group of Palo Altos. This northbound hub will be interesting if the server in the Subscriber VPC is a web server (it will be able to use the autoscaling features to meet the load).

This is why, from a Subscriber VPC point of view, I have only one specific route pointing to on-premises IP range. I need to keep a default route pointing to the IGW for the autoscale setup to work correctly.

I hope this is clear?

**Comment by originalwarby** [comment]
Vinch157,

The auto scaling solution requires the VM-Series firewalls to SNAT (AWS doesn't have a symmetric return option) so you won't need a default route to the IGW for the auto scaling to work.

The traffic inbound from the internet will appear to the server to be coming from the firewall on the local network.  Any traffic *initiated* by the server out to the internet, will take the default route to the VGW.  Therefore, no default route to the IGW is required.

HTH

**Comment by freimer** [comment]
The whole system appears to be keyed off of CreateTags and DeleteTags API calls, specifically for VPCs.  If you just delete a VPC it does not DeleteTags, so the automation never kicks off and the old CIDR is left in the SubscriberLocalDb DynamoDB table.  Did you delete the subscribingVpc tag on the old VPC and let it clean up, or just delete the VPC?  You may need to clean up manually.
